New 1.11 beta

Raycoms opened the pull request:
New beta

Review please

Raycoms committed:
Port1.11/fix/barbarians (#1479)

* Fix/barbarian spawns (#1477)

* Fix barbarian spawning, only calc building size once, decrease xp gain

* fix constructiontape

* restore and save them to NBT

* fix the javadoc

* fix codacity

* fix center returning null

* better calc

* Fix waypoint

* sonar fixes

* last adaptions

* fix merge errors

* some more adaptions

* fix the rotation

* codacity

* a bit more codacity

Raycoms committed:
Port/1.11/hotifxes (#1497)

Hotfixes several issues:

Crashes because of divide/0
Fixes baker furnaces
Fixes citizen job assignment
Fixes home assignment problems

Raycoms committed:
Whatafix/security measures (#1499) (#1500)

* security measures in tough times

* Fix despawning

* use the correct constant

* correctly spawn barbarians

Raycoms committed:
Port1.11/sounds n fix (#1516)

* Add male knight sounds

* Add female archer sounds

* codacity fix

* Fix/javadoc (#1515)

* fix all javadoc warnings

* delete file duplication

* Delete building on colony remove (#1514)

* Delete building on colony remove

* sort colonists by name

* fix not requesting item frames

* fix javadoc

* fix merge prblems

Raycoms committed:
Fix pasting in SMP + chisel handling (#1534) (#1536)

* update at a random moment

* chisels and bits support #1

* make paste work correctly in smp

* adapt properties

* fix update flag

* fix annots

* remove import

Raycoms committed:
Feature/builder gui (#1540) (#1542)

* new builder UI

* fix permission scepter problems

* Add baker sounds

* fix gui buttons

* fixed interaction sound and issue

* fix string

* fix infinite loop and add builder sounds

Raycoms committed:
Another bug fix attempt for the Permissions.View NPE.

Asherslab committed:
Port1.11/barbarian fixes (#1555)

* working

* Update MobEventsUtils.java

* Update Colony.java

Raycoms committed:
Feature/better avoid (#1558)

* avoid in random direction

* some small adaptions

* move away from location fix

* fix up all the going home and sleep stuff

* codacity issues

* fix problem with missing bed

* not necessary if

* hotfix bugfix NPE

Raycoms committed:
Port/standing logic (#1564)

* avoid in random direction

* Improving the standing logic of the Builder by adding some offsets.

Raycoms committed:
hotfix citizens without home running home! (#1570)

Asherslab committed:
init (#1576)

Raycoms committed:
Port1.11/colony commands (#1577)

* Create new colony commands to check for a close colony (#1572)

* Create new colony commands to check for a close colony

* fix desc

* fixes

* fix crash

* 1.11 adaptions

marchermans committed:
npe fix. (#1588)

Raycoms committed:
port bug fixes. (#1590)

Raycoms committed:
Add taiga style 1.11 (#1602)

* Add taiga style

* name fix

* fix item dupe bug

* fixy fix

* update schematics

* Several bugfixes

* fix bakery

* Rework the keep item completely (#1609)

* Rework the keep item completely

* Fix stack size

* Delete event handler on colony deletion

Asherslab committed:
fixed (#1621)

Raycoms committed:
Fix rack messup (#1616) (#1624)

* fixing rack messup

* Also request items in the hands of armor stands

Asherslab committed:
Done (#1620)

Raycoms committed:
Update README.md

Raycoms committed:
Feature/birch style (#1631) (#1640)

New birch style

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java:
```diff
@@ -36,12 +39,18 @@
+ *
  */
+@SuppressWarnings("squid:S2390")
 public abstract class AbstractBuilding
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 41 to the maximum authorized 30 or less. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1200)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') class "AbstractBuilding" has 68 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1448)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/sounds/KnightSounds.java:
```diff
@@ -0,0 +1,173 @@
+     * @param position position to play it at.
+     * @param isFemale the gender.
+     */
+    public void playInteractionSound(final World worldIn, final BlockPos position, final boolean isFemale)
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Provide multiple methods instead of using "isFemale" to determine which action to take. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2301)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/sounds/KnightSounds.java:
```diff
@@ -0,0 +1,173 @@
+     * @param isFemale   the gender.
+     * @param saturation the saturation.
+     */
+    public void playSaturationSound(final World worldIn, final BlockPos position, final boolean isFemale, final double saturation)
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Provide multiple methods instead of using "isFemale" to determine which action to take. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2301)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/util/SoundUtils.java:
```diff
@@ -124,4 +142,44 @@ public static void playSoundAtCitizenWithChance(@NotNull final World worldIn, @N
+                prefix = citizen.getWorkBuilding().getJobName();
+            }
+
+            if("GuardTower".equals(prefix) && citizen.getWorkBuilding() instanceof AbstractBuildingGuards)
```
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Use already-defined constant 'GUARD_TOWER' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java:
```diff
@@ -525,7 +544,48 @@ public Object handle(
+        }
+    }
+
+    public static class RackPlacementHandler implements IPlacementHandler
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 36 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2972)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java:
```diff
@@ -11,6 +11,7 @@
 import com.minecolonies.coremod.colony.buildings.BuildingWareHouse;
 import com.minecolonies.coremod.entity.EntityCitizen;
 import com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIStructure;
+import com.minecolonies.coremod.tileentities.TileEntityRack;
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused import 'com.minecolonies.coremod.tileentities.TileEntityRack'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AUselessImportCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/permissions/Permissions.java:
```diff
@@ -938,7 +938,7 @@ public boolean hasPermission(final UUID id, @NotNull final Action action)
         {
             return (rank == Rank.OWNER && action != Action.GUARDS_ATTACK)
-                     || Utils.testFlag(permissions.get(rank), action.getFlag());
+                     || (permissions != null && action != null && permissions.containsKey(rank) && Utils.testFlag(permissions.get(rank), action.getFlag()));
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2589)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBarracks.java:
```diff
@@ -111,21 +113,24 @@ public void onUpgradeComplete(final int newLevel)
 
         EnumFacing facing = EnumFacing.NORTH;
+
+        final int offset = getStyle().toLowerCase(Locale.ENGLISH).contains("birch") ? TOWER_OFFSET + 1: TOWER_OFFSET;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add parentheses to make the operator precedence explicit. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS864)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java:
```diff
@@ -225,8 +251,11 @@ private void addHomelessCitizens()
             {
-                // The citizen can move to this hut to improve conditions
-                citizen.getHomeBuilding().removeCitizen(citizen);
+                if(citizen.getHomeBuilding() instanceof BuildingHome)
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1066)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java:
```diff
@@ -163,6 +188,7 @@ public void removeCitizen(@NotNull final CitizenData citizen)
         {
             citizen.setHomeBuilding(null);
             residents.remove(citizen);
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "remove()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2250)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/event/ClientEventHandler.java:
```diff
@@ -101,7 +101,7 @@ public void renderWorldLastEvent(@NotNull final RenderWorldLastEvent event)
 
             final ColonyView colony = ColonyManager.getClosestColonyView(world, player.getPosition());
-            if(colony != null && colony.getPermissions().hasPermission(player, Action.ACCESS_HUTS))
+            if(colony != null && player != null && colony.getPermissions().hasPermission(player, Action.ACCESS_HUTS))
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2589)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java:
```diff
@@ -324,7 +300,8 @@ public void onWorldTick(@NotNull final TickEvent.WorldTickEvent event)
 
         // If we have no active worker, grab one from the Colony
         // TODO Maybe the Colony should assign jobs out, instead?
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1135)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/Colony.java:
```diff
@@ -212,6 +241,11 @@
     private double overallHappiness = 5;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 5 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS109)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/Colony.java:
```diff
@@ -46,14 +48,16 @@
  * This class describes a colony and contains all the data and methods for
  * manipulating a Colony.
  */
 public class Colony implements IColony
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 50 to the maximum authorized 30 or less. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1200)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') class "Colony" has 99 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1448)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/util/ColonyUtils.java:
```diff
@@ -44,4 +48,27 @@ public static boolean hasNewSubscribers(@NotNull final Set<EntityPlayerMP> oldSu
+     * @param isMirrored if its mirrored.
+     * @return a tuple with the required corners.
+     */
+    public static Tuple<Tuple<Integer, Integer>, Tuple<Integer, Integer>> calculateCorners(
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Provide multiple methods instead of using "isMirrored" to determine which action to take. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2301)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java:
```diff
@@ -3,6 +3,7 @@
 import com.minecolonies.api.util.CompatibilityUtils;
 import com.minecolonies.api.util.InventoryUtils;
 import com.minecolonies.api.util.ItemStackUtils;
+import com.minecolonies.api.util.constant.Constants;
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused import 'com.minecolonies.api.util.constant.Constants'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AUselessImportCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/CitizenData.java:
```diff
@@ -233,6 +233,70 @@ public void markDirty()
     }
 
+    @Override
+    public boolean equals(final Object o)
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 11, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1142)

MinecoloniesSonar commented:
SonarQube analysis reported 209 issues
* ![CRITICAL][CRITICAL] 7 critical
* ![MAJOR][MAJOR] 107 major
* ![MINOR][MINOR] 80 minor
* ![INFO][INFO] 15 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [CitizenData.java#L190](https://github.com/Minecolonies/minecolonies/blob/068e80fc10c65f8ebc01e11910c1d41d9c93120a/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L190): Use already-defined constant 'TAG_SKILLS' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L191](https://github.com/Minecolonies/minecolonies/blob/068e80fc10c65f8ebc01e11910c1d41d9c93120a/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L191): Use already-defined constant 'TAG_SKILL_STRENGTH' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L192](https://github.com/Minecolonies/minecolonies/blob/068e80fc10c65f8ebc01e11910c1d41d9c93120a/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L192): Use already-defined constant 'TAG_SKILL_STAMINA' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L193](https://github.com/Minecolonies/minecolonies/blob/068e80fc10c65f8ebc01e11910c1d41d9c93120a/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L193): Use already-defined constant 'TAG_SKILL_SPEED' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L194](https://github.com/Minecolonies/minecolonies/blob/068e80fc10c65f8ebc01e11910c1d41d9c93120a/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L194): Use already-defined constant 'TAG_SKILL_INTELLIGENCE' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L195](https://github.com/Minecolonies/minecolonies/blob/068e80fc10c65f8ebc01e11910c1d41d9c93120a/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L195): Use already-defined constant 'TAG_SKILL_DEXTERITY' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [BlockMinecoloniesRack.java#L131](https://github.com/Minecolonies/minecolonies/blob/068e80fc10c65f8ebc01e11910c1d41d9c93120a/src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java#L131): Reduce the number of returns of this method 7, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [BlockMinecoloniesRack.java#L131](https://github.com/Minecolonies/minecolonies/blob/068e80fc10c65f8ebc01e11910c1d41d9c93120a/src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java#L131): Add the missing @Deprecated annotation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
1. ![MAJOR][MAJOR] [BlockMinecoloniesRack.java#L381](https://github.com/Minecolonies/minecolonies/blob/068e80fc10c65f8ebc01e11910c1d41d9c93120a/src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java#L381): Reduce this class from 51 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [WindowBuildTool.java#L36](https://github.com/Minecolonies/minecolonies/blob/068e80fc10c65f8ebc01e11910c1d41d9c93120a/src/main/java/com/minecolonies/coremod/client/gui/WindowBuildTool.java#L36): class "WindowBuildTool" has 38 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1448)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:
```diff
@@ -859,24 +882,31 @@ public void onDeath(final DamageSource par1DamageSource)
+    @Override
+    public EnumActionResult applyPlayerInteraction(final EntityPlayer player, final Vec3d vec, final EnumHand hand)
+    {
+        SoundUtils.playInteractionSoundAtCitizenWithChance(CompatibilityUtils.getWorld(this), this.getPosition(), 100, this);
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 100 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS109)

RVOtakuMike committed:
Ported 2.5 commands (#1643)

Ported CheckForAutoDeletsCommand, MakeNotAutoDeletable, and my changes to DeleteColonyCommand

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/generalcommands/CheckForAutoDeletesCommand.java:
```diff
@@ -0,0 +1,114 @@
+        {
+            final Colony colony = colonies.get(index);
+
+            if (colony.canBeAutoDeleted() && Configurations.gameplay.autoDeleteColoniesInHours != 0 && colony.getLastContactInHours() >= Configurations.gameplay.autoDeleteColoniesInHours)
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this 187 characters long line (which is greater than 180 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS00103)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/ColonyManager.java:
```diff
@@ -187,22 +186,28 @@ public static void deleteColony(final int id)
-                        Log.getLogger().info("Found Block, deleting " + world.getBlockState(location).getBlock());
-                        world.setBlockToAir(location);
+                Log.getLogger().info("Removing buildings for " + id);
+                for (final AbstractBuilding building : new ArrayList<>(colony.getBuildings().values())) {
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/ColonyManager.java:
```diff
@@ -187,22 +186,28 @@ public static void deleteColony(final int id)
+                    Log.getLogger().info("Delete Building at " + location);
+                    building.deconstruct();
+                    building.destroy();
+                    for (final World world : colonyWorlds) {
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/ColonyManager.java:
```diff
@@ -187,22 +186,28 @@ public static void deleteColony(final int id)
+                    building.deconstruct();
+                    building.destroy();
+                    for (final World world : colonyWorlds) {
+                        if (world.getBlockState(location).getBlock() instanceof AbstractBlockHut) {
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/ColonyManager.java:
```diff
@@ -166,14 +167,12 @@ public static void markDirty()
      * @param id the colonies id.
      */
-    public static void deleteColony(final int id)
+    public static void deleteColony(final int id, final boolean canDestroy)
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Provide multiple methods instead of using "canDestroy" to determine which action to take. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2301)

MinecoloniesSonar commented:
SonarQube analysis reported 215 issues
* ![CRITICAL][CRITICAL] 7 critical
* ![MAJOR][MAJOR] 109 major
* ![MINOR][MINOR] 84 minor
* ![INFO][INFO] 15 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [CitizenData.java#L190](https://github.com/Minecolonies/minecolonies/blob/6d3c8c2b2eec7099026cb340d2398280d9b230b5/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L190): Use already-defined constant 'TAG_SKILLS' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L191](https://github.com/Minecolonies/minecolonies/blob/6d3c8c2b2eec7099026cb340d2398280d9b230b5/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L191): Use already-defined constant 'TAG_SKILL_STRENGTH' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L192](https://github.com/Minecolonies/minecolonies/blob/6d3c8c2b2eec7099026cb340d2398280d9b230b5/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L192): Use already-defined constant 'TAG_SKILL_STAMINA' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L193](https://github.com/Minecolonies/minecolonies/blob/6d3c8c2b2eec7099026cb340d2398280d9b230b5/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L193): Use already-defined constant 'TAG_SKILL_SPEED' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L194](https://github.com/Minecolonies/minecolonies/blob/6d3c8c2b2eec7099026cb340d2398280d9b230b5/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L194): Use already-defined constant 'TAG_SKILL_INTELLIGENCE' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L195](https://github.com/Minecolonies/minecolonies/blob/6d3c8c2b2eec7099026cb340d2398280d9b230b5/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L195): Use already-defined constant 'TAG_SKILL_DEXTERITY' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [BlockMinecoloniesRack.java#L131](https://github.com/Minecolonies/minecolonies/blob/6d3c8c2b2eec7099026cb340d2398280d9b230b5/src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java#L131): Reduce the number of returns of this method 7, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [BlockMinecoloniesRack.java#L131](https://github.com/Minecolonies/minecolonies/blob/6d3c8c2b2eec7099026cb340d2398280d9b230b5/src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java#L131): Add the missing @Deprecated annotation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
1. ![MAJOR][MAJOR] [BlockMinecoloniesRack.java#L381](https://github.com/Minecolonies/minecolonies/blob/6d3c8c2b2eec7099026cb340d2398280d9b230b5/src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java#L381): Reduce this class from 51 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [WindowBuildTool.java#L36](https://github.com/Minecolonies/minecolonies/blob/6d3c8c2b2eec7099026cb340d2398280d9b230b5/src/main/java/com/minecolonies/coremod/client/gui/WindowBuildTool.java#L36): class "WindowBuildTool" has 38 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1448)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'
